public/app/plugins/datasource/loki/components/AnnotationQueryEditor.tsx

```diff
@@ -0,0 +1,50 @@
+// Libraries
+import React, { memo } from 'react';
+
+// Types
+import { DataSourceStatus } from '@grafana/ui';
+import { LokiQuery } from '../types';
+import { useLokiSyntax } from './useLokiSyntax';
+import { LokiQueryFieldForm } from './LokiQueryFieldForm';
+
+export const LokiAnnotationQueryEditor = memo(function LokiAnnotationQueryEditor(props: any) {
```

Comment:
should props not be typed here, think we should use normal `QueryEditorProps<T>`

Reply:
I added the Prop types but it is a bit different from `QueryEditorProps<T>` as only a subset is passed from the Angular controller and others are created here in this component.